src/isomorphic/classic/element/ReactElement.js

```diff
@@ -146,6 +147,62 @@ ReactElement.createElement = function(type, config, children) {
     }
   }
 
+  if (__DEV__) {
+
+    var reactProps = {
+      children: true,
+      dangerouslySetInnerHTML: true,
+      key: true,
+      ref: true,
+    };
+
+    var warnedProperties = {};
+
+    for (propName in config) {
+      var lowerCasedName = propName.toLowerCase();
+
+      if (!DOMProperty.properties.hasOwnProperty(propName) && 
+          !DOMProperty.isCustomAttribute(propName) && 
+          !(reactProps.hasOwnProperty(name) && reactProps[name] || 
+            warnedProperties.hasOwnProperty(name) && warnedProperties[name]) && 
+          EventPluginRegistry.possibleRegistrationNames[lowerCasedName] !== propName) {
+
+        warnedProperties[name] = true;
+
+        var standardName = (
+          DOMProperty.isCustomAttribute(lowerCasedName) ?
+            lowerCasedName :
+          DOMProperty.getPossibleStandardName.hasOwnProperty(lowerCasedName) ?
+            DOMProperty.getPossibleStandardName[lowerCasedName] :
+            null
+        );
+        
+        // For now, only warn when we have a suggested correction. This prevents
+        // logging too much when using transferPropsTo.
+        warning(
+          standardName == null,
+          'Unknown DOM property %s. Did you mean %s?',
+          propName,
+          standardName
+        );
+
+        var registrationName = (
+          EventPluginRegistry.possibleRegistrationNames.hasOwnProperty(
+            lowerCasedName
+          ) ?
+            EventPluginRegistry.possibleRegistrationNames[lowerCasedName] :
+            null
+        );
+
+        warning(
```

Comment:
I think we want to use the “devtool” API for the new warnings. You can check how `ReactDOMUnknownPropertyDevtool` currently works for the context—it just needs to be called during the element creation.